crates/cheatnet/src/cheatcodes/spoof.rs

```diff
@@ -0,0 +1,47 @@
+use crate::CheatnetState;
+use cairo_felt::Felt252;
+use starknet_api::core::ContractAddress;
+
+pub struct TxInfoMock {
+    pub version: Option<Felt252>,
+    pub account_contract_address: Option<Felt252>,
+    pub max_fee: Option<Felt252>,
+    pub signature: Option<Vec<Felt252>>,
+    pub transaction_hash: Option<Felt252>,
+    pub chain_id: Option<Felt252>,
+    pub nonce: Option<Felt252>,
+}
+
+impl CheatnetState {
+    #[allow(clippy::too_many_arguments)]
+    pub fn start_spoof(
+        &mut self,
+        contract_address: ContractAddress,
+        version: Option<Felt252>,
+        account_contract_address: Option<Felt252>,
+        max_fee: Option<Felt252>,
+        signature: Option<Vec<Felt252>>,
+        transaction_hash: Option<Felt252>,
+        chain_id: Option<Felt252>,
+        nonce: Option<Felt252>,
```

Comment:
This isn't necessarly a better approach, but what if `TxInfoMock` implemented `Default` trait, and this function accepted this struct?

This way in case we just want to change one value (e.g. in tests) we could do something along the lines of 
```rust
let mocked = TxInfoMock { version: Some(Felt252::from(1), ..TxInfoMock::default() };
start_spoof(mocked);
```

Reply:
In the spirit of https://github.com/foundry-rs/starknet-foundry/issues/506 `Cheatnet` `start_mock` accepts individual fields instead of the `TxInfoMock` structure.  So in the current implementation, I don't see clear benefits of using `TxInfoMock::default()`

Reply:
To me the benefit is mostly readability in tests + not having to pass x number of arguments if only one changes. But if you think your approach is better, we can keep it as it is.